### PR TITLE
REST API Source

### PR DESCRIPTION
**Pull request by cgardens**
## What
* I thought this would be a cool source. The idea is that any http request can now be a source. Yes, we will always prefer to a source customized for an API if we have it. But if it doesn't exist and you have a case where you just want dump the contents of an HTTP request into a database, this will do it. It's a pretty simple source, but I think this could be really in addressing weird long tail cases. Hopefully it's a high leverage sort of thing.
  * This is a little weird when we only support full refresh and I think will make more sense when we have append.
* Also a bit of an excuse to try writing a python source from scratch. 😉 
* **UPDATE** - the original PR did not include adding the link to the docs from the gitbook index. This [followup PR](https://github.com/airbytehq/airbyte/pull/801) fixes that.

### CODE REVIEW

File: airbyte-integrations/connectors/source-rest-api/airbyte_protocol [hunk 1]
```diff
@@ -0,0 +1 @@
+/Users/charles/code/airbyte/airbyte-integrations/bases/airbyte-protocol/airbyte_protocol
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-rest-api/airbyte_protocol:1]: use relative path

File: airbyte-integrations/connectors/source-rest-api/base_python [hunk 2]
```diff
@@ -0,0 +1 @@
+/Users/charles/code/airbyte/airbyte-integrations/bases/base-python/base_python
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-rest-api/base_python:1]: same here

File: airbyte-integrations/connectors/source-rest-api/integration_tests/standard_source_test.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+    def get_spec(self) -> ConnectorSpecification:
+        raw_spec = pkgutil.get_data(self.__class__.__module__.split(".")[0], "spec.json")
+        return ConnectorSpecification.parse_obj(json.loads(raw_spec))
+
+    def get_config(self) -> object:
+        return {
+            "http_method": "get",
+            "url": "http://api.bart.gov/api/route.aspx?cmd=routes&key=MW9S-E7SL-26DU-VV8V&json=y",
+            "headers": '{"Content-Typ": "application/json"}',
```
Comment (michel-tricot) [inline airbyte-integrations/connectors/source-rest-api/integration_tests/standard_source_test.py:44]: typo in the header

**Review by michel-tricot — APPROVED** [review]
AMAZING!

File: airbyte-integrations/connectors/source-rest-api/source_rest_api/source.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        r = self._make_request(config_container.rendered_config)
+        if r.status_code != 200:
+            raise Exception(f"Request failed. {r.text}")
+
+        # need to eagerly fetch the json.
+        message = AirbyteMessage(
+            type=Type.RECORD,
+            record=AirbyteRecordMessage(stream=SourceRestApi.STREAM_NAME, data=r.json(), emitted_at=int(datetime.now().timestamp()) * 1000),
+        )
+        return (m for m in [message])
```
Comment (cgardens) [inline airbyte-integrations/connectors/source-rest-api/source_rest_api/source.py:63]: boh? there's got to be a better way to do a singleton generator.

File: airbyte-integrations/connectors/source-rest-api/airbyte_protocol (hunk 1 above)
Comment (cgardens) [reply airbyte-integrations/connectors/source-rest-api/airbyte_protocol:1]: oh interesting. this was autogenerated. but yes. definitely relative path. will try to figure out why template is spitting out absolute path.

Comment (cgardens) [reply airbyte-integrations/connectors/source-rest-api/airbyte_protocol:1]: i fixed this in the template https://github.com/airbytehq/airbyte/pull/783. will fix here now.

File: airbyte-integrations/connectors/source-rest-api/source_rest_api/source.py (hunk 4 above)
Comment (eugene-kulak) [reply airbyte-integrations/connectors/source-rest-api/source_rest_api/source.py:63]: `yield message`

Comment (michel-tricot) [reply airbyte-integrations/connectors/source-rest-api/source_rest_api/source.py:63]: Thanks Eugene, we will fix it. Or if you feel like it you can do a quick PR

**Pull request merged**